x/eibc/types/tx.go

```diff
@@ -37,7 +42,7 @@ func (msg *MsgFulfillOrder) GetSignBytes() []byte {
 }
 
 func (m *MsgFulfillOrder) ValidateBasic() error {
-	if m.OrderId == "" {
+	if m.OrderId == "" || len(m.OrderId) > maxLengthOfOrderID {
```

Comment:
as we set the orderId progtramtticly, I think it can be more strictly checked
by reversing the encode func

```
func isValidHash(hashString string) bool {
    hashBytes, err := hex.DecodeString(hashString)
    if err != nil {
        // The string is not a valid hexadecimal string
        return false
    }
    // SHA-256 hashes are 32 bytes long
    return len(hashBytes) == 32
}
```